config/routes.rb

```diff
@@ -17,6 +17,8 @@
   end
 
   resources :nodes, only: [:index, :show, :destroy] do
+    post :assign
```

Comment:
i tried to keep the api generic, but i am not 100% sure if that makes sense. do we want to assign e.g. a minion role explicitly?